pkg/v1/remote/referrers.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+)
+
+// Referrers returns a list of descriptors that refer to the given manifest digest.
+//
+// The subject manifest doesn't have to exist in the registry for there to be descriptors that refer to it.
+func Referrers(d name.Digest, options ...Option) (*v1.IndexManifest, *ReferrersNextPage, error) {
+	o, err := makeOptions(d.Context(), options...)
+	if err != nil {
+		return nil, nil, err
+	}
+	f, err := makeFetcher(d, o)
+	if err != nil {
+		return nil, nil, err
+	}
+	return f.fetchReferrers(o.context, o.filter, d)
+}
+
+// TODO: Implement this once real referrers API endpoint is used
+type ReferrersNextPage struct{}
+
+// TODO: Implement this once real referrers API endpoint is used
+func (nextPage *ReferrersNextPage) Referrers() (*v1.IndexManifest, *ReferrersNextPage, error) {
+	panic("not yet implemented")
+}
```

Comment:
open to other suggestions - picturing the case with pagination, where the `ReferrersNextPage` has stored the contents of the `Link` header, then called like this for example:

```go
var index *v1.IndexManifest
var next *remote.ReferrersNextPage
var err error
for next == nil {
	if index == nil { 
		index, next, err = Referrers(d)      // First time through the loop
	} else { 
		index, next, err = next.Referrers(d) // If we got here, we are on the next page
	}
	if err != nil {
		return err
	}
	processIndex(index)
}
```

Reply:
after discussion w/ @imjasonh, the paging can be hidden from the client (similar to tag listing)

If needed later, can add an option to limit it etc